Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/SocialLogin.hs

```diff
@@ -139,21 +139,27 @@ postSocialUpdateProfile ::
     API.Types.UI.SocialLogin.SocialUpdateProfileReq ->
     Environment.Flow Kernel.Types.APISuccess.APISuccess
   )
-postSocialUpdateProfile (mbPersonId, _, _) req = do
-  case mbPersonId of
-    Nothing -> throwError $ InternalError "Not Implemented for dashboard"
-    Just personId -> do
-      encNewPhoneNumber <- encrypt `mapM` req.mobileNumber
-      person <-
-        PQ.findById personId
-          >>= fromMaybeM (PersonDoesNotExist personId.getId)
-      let updPerson =
-            person
-              { SP.mobileCountryCode = req.mobileCountryCode,
-                SP.mobileNumber = encNewPhoneNumber,
-                SP.unencryptedMobileNumber = req.mobileNumber,
-                SP.firstName = fromMaybe person.firstName req.firstName,
-                SP.lastName = req.lastName <|> person.lastName
-              }
-      PQ.updatePersonDetails updPerson
-      pure Kernel.Types.APISuccess.Success
+postSocialUpdateProfile (mbPersonId, merchantId, _) req = do
+  personId <- maybe (throwError $ InternalError "Not Implemented for dashboard") pure mbPersonId
+  encNewPhoneNumber <- mapM encrypt req.mobileNumber
+  person <- PQ.findById personId >>= fromMaybeM (PersonDoesNotExist personId.getId)
+  case req.mobileNumber of
+    Just mobileNumber -> do
+      mobileNumberHash <- getDbHash mobileNumber
+      let countryCode = fromMaybe "+91" req.mobileCountryCode
+      PQ.findByMobileNumberAndMerchantId countryCode mobileNumberHash merchantId >>= \case
```

Comment:
Lets use this function instead of creating a new one `findByMobileNumberAndMerchantAndRole`